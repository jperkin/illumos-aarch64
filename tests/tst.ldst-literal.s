/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.3.5 Load register (literal)
 *
 * C5.6.144 PRFM (literal) is handled by tst.ldst-prfm.s
 */
.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
1:
	ldr	w0,  2f
	ldr	xzr, 1b
2:
	ldr	s0,  2b
	ldr	d30, 3f
	ldr	q31, 1b
3:
	ldrsw	x30, 2b
.size libdis_test, [.-libdis_test]
