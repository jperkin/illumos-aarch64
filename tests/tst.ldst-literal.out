    libdis_test:      40 00 00 18  ldr w0, 8 	<libdis_test+0x8>
    libdis_test+0x4:  ff ff ff 58  ldr xzr, 0 	<libdis_test>
    libdis_test+0x8:  00 00 00 1c  ldr s0, 8 	<libdis_test+0x8>
    libdis_test+0xc:  5e 00 00 5c  ldr d30, 14 	<libdis_test+0x14>
    libdis_test+0x10: 9f ff ff 9c  ldr q31, 0 	<libdis_test>
    libdis_test+0x14: be ff ff 98  ldrsw x30, 8 	<libdis_test+0x8>
